internal/data/common.go

```diff
@@ -140,6 +142,136 @@ func NewAttributes(m AttributesMap, droppedCount uint32) Attributes {
 	return Attributes{m, droppedCount}
 }
 
+// StringKeyValue stores a key and AttributeValue pair.
+type StringKeyValue struct {
+	orig *otlpcommon.StringKeyValue
+}
+
+// NewStringKeyValue creates a new StringKeyValue with the given key and value.
+func NewStringKeyValue(k string, v string) StringKeyValue {
+	return StringKeyValue{&otlpcommon.StringKeyValue{Key: k, Value: v}}
+}
+
+// Key returns the key associated with this StringKeyValue.
+func (akv StringKeyValue) Key() string {
+	return akv.orig.Key
+}
+
+// Value returns the value associated with this StringKeyValue.
+func (akv StringKeyValue) Value() string {
+	return akv.orig.Value
+}
+
+// SetValue replaces the value associated with this StringKeyValue.
+func (akv StringKeyValue) SetValue(v string) {
+	akv.orig.Value = v
+}
+
+// StringMap stores a map of attribute keys to values.
+type StringMap struct {
+	orig *[]*otlpcommon.StringKeyValue
+}
+
+func newStringMap(orig *[]*otlpcommon.StringKeyValue) StringMap {
+	return StringMap{orig}
+}
+
+// NewStringMap creates a new StringMap from the given map[string]AttributeValue.
+func NewStringMap(attrMap map[string]string) StringMap {
+	if len(attrMap) == 0 {
+		var orig []*otlpcommon.StringKeyValue
+		orig = nil
+		return StringMap{&orig}
+	}
+	origs := make([]otlpcommon.StringKeyValue, len(attrMap))
+	wrappers := make([]*otlpcommon.StringKeyValue, len(attrMap))
+
+	ix := 0
+	for k, v := range attrMap {
+		wrappers[ix] = &origs[ix]
+		wrappers[ix].Key = k
+		StringKeyValue{wrappers[ix]}.SetValue(v)
```

Comment:
Why not just:
```go
wrappers[ix].Value = v
```
This is more obvious, I think.

Reply:
Done.